invenio_rdm_records/resources/serializers/signposting/schema.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Northwestern University.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Signposting schemas."""
+
+import idutils
+from marshmallow import Schema, fields, missing
+
+from ...urls import download_url_for
+from ..utils import get_vocabulary_props
+
+
+class LandingPageSchema(Schema):
+    """Schema for serialization of typed links pertaining to the landing page.
+
+    Serialization input is a whole record dict projection.
+    """
+
+    anchor = fields.Method(serialize="serialize_anchor")
+    author = fields.Method(serialize="serialize_author")
+    cite_as = fields.Method(data_key="cite-as", serialize="serialize_cite_as")
+    describedby = fields.Method(serialize="serialize_describedby")
+    item = fields.Method(serialize="serialize_item")
+    license = fields.Method(serialize="serialize_license")
+    type = fields.Method(serialize="serialize_type")
+
+    def serialize_anchor(self, obj, **kwargs):
+        """Seralize to landing page URL."""
+        return obj["links"]["self_html"]
+
+    def serialize_author(self, obj, **kwargs):
+        """Serialize author(s).
+
+        For now, the first linkable identifier is taken.
+        """
+
+        def pick_linkable_id(identifiers):
+            for id_dict in identifiers:
+                url = idutils.to_url(
+                    id_dict["identifier"], id_dict["scheme"], url_scheme="https"
+                )
+                if url:
+                    return url
+                else:
+                    continue
+            return None
+
+        metadata = obj["metadata"]
+        result = [
+            {"href": pick_linkable_id(c["person_or_org"].get("identifiers", []))}
+            for c in metadata.get("creators", [])
+        ]
+        result = [r for r in result if r["href"]]
+        return result or missing
+
+    def serialize_cite_as(self, obj, **kwargs):
+        """Serialize cite-as."""
+        identifier = obj.get("pids", {}).get("doi", {}).get("identifier")
```

Comment:
Probably we need to move to the last version of the record (api/records/7ze6y-5fa68/versions/latest) in order  to get pids object since it depends on  the version of the record
 I wonder if  we can add other pids in addition to the  DOI such oai-pmh ?

Reply:
Ah, per the spec for the cite-as field:

> Provide a link that has the persistent identifier of the scholarly object (e.g. DOI expressed as an HTTP URI) as target. Note that dereferencing the target of a link with a cite-as relation type must lead to the landing page that provided that cite-as link. 

, the signposting is per scholarly object (i.e. version), so getting the current version's DOI as we do is right. Different versions are cited differently and that's good since we can refer to a specific version. 

The cardinality of cite-as is 1 and its link must follow the conditions above (persistent and redirects back to this page) so an oai-pmh link probably wouldn't do.

It would be good to bring up the notion of versions to the spec authors. FAIR profiling mentions the landing page, resource and metadata, but doesn't mention inter-version connections.

Reply:
We have to check if the record didn't have a DOI identifier but others PIDs were  mentioned and to decide which identifier we will chose!